packages/model-viewer/src/features/scene-graph/texture.ts

```diff
@@ -36,15 +37,25 @@ export class Texture extends ThreeDOMElement implements TextureInterface {
 
   constructor(
       onUpdate: () => void,
-      sampler: Sampler,
-      image: Image,
-      texture: ThreeTexture|null,
+      threeTexture: ThreeTexture|null,
       gltfTexture: GLTFTexture|null,
+      gltfSampler: GLTFSampler|null,
+      gltfImage: GLTFImage|null,
   ) {
     super(
-        onUpdate, gltfTexture, new Set<ThreeTexture>(texture ? [texture] : []));
-    this[$sampler] = sampler;
-    this[$image] = image;
+        onUpdate,
+        gltfTexture,
+        new Set<ThreeTexture>(threeTexture ? [threeTexture] : []));
+
+    this[$sampler] = new Sampler(onUpdate, threeTexture, gltfSampler);
+    this[$image] = new Image(onUpdate, threeTexture, gltfImage);
+  }
+
+  // Helpter method creates a Texture from a ThreeTexture that is not bound to
+  // any material.
+  static[$createTexture](threeTexture: ThreeTexture, onUpdate: () => void):
+      Texture {
+    return new Texture(onUpdate, threeTexture, null, null, null);
```

Comment:
Well, that's kind of trivial now. Should we just remove this and call the constructor directly?

Reply:
Done.